.github/workflows/test.yml

```diff
@@ -5,6 +5,10 @@ on:
   workflow_dispatch:
   workflow_call:
 
+concurrency:
+  group: ${{ github.ref }}
+  cancel-in-progress: ${{ github.ref != 'refs/heads/master' }}
```

Comment:
if a newer commits occurs on branch, cancel last job in progress.  (would only occur if commits are faster than 4 minutes)